File: eshop-jordan/src/main/java/model/Fournisseur.java
package model;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.OneToMany;

@Entity
@DiscriminatorValue("supplier")
public class Fournisseur extends Personne{

	
	@Column(name="company",length = 25)
	private String societe;
	
	
	@OneToMany(mappedBy = "fournisseur")
	private transient List<Produit> produits;

	public Fournisseur() {
	}
	
	public Fournisseur(String nom, String prenom, String societe) {
		super(nom, prenom);
		this.societe = societe;
	}

	public String getSociete() {
		return societe;
	}

	public void setSociete(String societe) {
		this.societe = societe;
	}
	
	

	public List<Produit> getProduits() {
		return produits;
	}

	public void setProduits(List<Produit> produits) {
		this.produits = produits;
	}

	@Override
	public String toString() {
		return "Fournisseur [id=" + id + ", nom=" + nom + ", prenom=" + prenom + ", adresse=" + adresse + ", societe="
				+ societe + "]";
	}
	
	
	
}

File: olivier/ANGULAR/eshop/src/app/eshop/model/fournisseur.ts
import { Adresse } from './adresse';
import { Personne } from './personne';
export class Fournisseur extends Personne {
  public get societe(): string | undefined {
    return this._societe;
  }
  public set societe(value: string | undefined) {
    this._societe = value;
  }

  constructor(
    _id?: number,
    _nom?: string,
    _prenom?: string,
    _adresse?: Adresse,
    private _societe?: string
  ) {
    super(_id, _nom, _prenom, _adresse);
  }
}
